docs/glossary.md

```diff
@@ -0,0 +1,71 @@
+# OpenTofu glossary
+This document is intended for anyone who wants to gain more knowledge about the terms and vocabulary used to talk about different concepts in OpenTofu.
+
+> [!NOTE]
+> This was created with the intent of gathering more knowledge over time.
+> The state that you find this in right now could be incomplete. 
+> Once you discover and learn about a new concept that would benefit others,
+> feel free to open a PR to update this.
+
+> [!NOTE]
+> When adding new content to this document, try to place it in such a way to match the alphabetical order of the already existing content.
+> 
+> Optionally, also add a reference link if possible (GitHub conversation, issue, other docs, etc).
+
+## Attribute/argument/field
+* Attribute - a named key inside an object type.
+* Argument - a name used for an individual setting inside a configuration block.
+
+It is recommended to avoid using popular programming language terms such as "field" or "property" to describe an element of an object in OpenTofu.
+
+Reference: [link](./diagnostics.md#diagnostic-description-writing-style)
+
+## Data sources/data resource
+* "Data resource" refers to a block of type data, and the associated object it declares.
+* "Data source" is the remote thing that the data resource reads from.
+* "Data resource type" is what is represented by the first label in a data block header, and the associated declarations and code for it in the provider plugin.
+
+Reference: [link](https://github.com/opentofu/opentofu/pull/3389#discussion_r2440264786)
+
+## Diagnostic
+"Diagnostics" is the general term we use to describe the error and warning
+messages that OpenTofu returns when there are problems with the configuration,
+or when interactions with external systems fail.
+
+Reference: [link](./diagnostics.md)
+
+## Expression
+Expressions refer to or compute values within a configuration. The simplest expressions are just literal values, like "hello" or 5, but the OpenTofu language also allows more complex expressions such as references to data exported by resources, arithmetic, conditional evaluation, and a number of built-in functions.
+
+Reference: [link](https://opentofu.org/docs/language/expressions/)
+
+## Evaluation context
+A set of already known functions, input values, local values, resources, etc. that is used to evaluate an expression that can reference any of the concepts listed above.
+
+The list of concepts above, in the context of HCL evaluation, are called [variables](#variable-hcl).
```

Comment:
Would be good to note that this is an HCL eval context, not a tofu EvalContext, the catch all structure that is facilitates evaluation in the current tofu graph structure.